utilities/write_batch_with_index/write_batch_with_index_internal.cc

```diff
@@ -94,9 +94,11 @@ int WriteBatchEntryComparator::operator()(
     return -1;
   }
 
-  if (entry1->offset == WriteBatchIndexEntry::kFlagMin) {
+  if (entry1->offset == WriteBatchIndexEntry::kFlagMin &&
```

Comment:
Can we add inline comments of why this special case mean and why we are returning -1?

Reply:
Also would be great to add some inline comments above kFlagMin of when we make use of it? For what kind of entry we would set this flag?